python/lsst/meas/extensions/gaap/gaap.py

```diff
@@ -0,0 +1,319 @@
+    """
+    def __init__(self, errors: dict[str, Exception], *args, **kwds):
+        message = "Problematic scaling factors = "
+        message += ", ".join(errors)
+        message += " Errors: "
+        message += " | ".join(set(msg.__repr__() for msg in errors.values()))  # msg.cpp.what() misses type
+        super().__init__(message, *args, **kwds)
+
+
+class BaseGaapFluxConfig(lsst.meas.base.BaseMeasurementPluginConfig):
+    """Configuration parameters for Gaussian Aperture and PSF (GAaP) plugin.
+    """
+    def _greaterThanOrEqualToUnity(x: float) -> bool:  # noqa: N805
+        """Returns True if the input ``x`` is greater than 1.0, else False.
+        """
+        return x >= 1
+
+    def _isOdd(x: int) -> bool:  # noqa: N805
+        """Returns True if the input ``x`` is positive and odd, else False.
+        """
+        return (x%2 == 1) & (x > 0)
+
+    sigmas = ListField(dtype=float, default=[4.0, 5.0],
+                       doc=("List of sigmas (in pixels) of circular Gaussian apertures to apply on "
+                            "pre-seeing galaxy images. These should be somewhat larger than the PSF "
+                            "(determined by ``scalingFactors``) to avoid measurement failures."
+                            ),
+                       )
+
+    scalingFactors = ListField(dtype=float, default=[1.15], itemCheck=_greaterThanOrEqualToUnity,
+                               doc=("List of factors with which the seeing should be scaled to obtain the "
+                                    "sigma values of the target Gaussian PSF. The factor should not be less "
+                                    "than unity to avoid the PSF matching task to go into deconvolution mode "
+                                    "and should ideally be slightly greater than unity."
+                                    ),
+                               )
+
+    modelPsfMatch = ConfigurableField(doc="PSF Gaussianization Task", target=ModelPsfMatchTask)
+
+    modelPsfDimension = Field(dtype=int, default=65, check=_isOdd,
+                              doc=("The dimensions (width and height) of the target PSF image in pixels. "
+                                   "Must be odd."
+                                   ),
+                              )
+
+    def setDefaults(self) -> None:
+        # TODO: DM-27482 might change these values.
+        self.modelPsfMatch.kernel.active.alardNGauss = 1
+        self.modelPsfMatch.kernel.active.alardDegGaussDeconv = 1
+        self.modelPsfMatch.kernel.active.alardDegGauss = [8]
+        self.modelPsfMatch.kernel.active.alardSigGauss = [5.0]
+        self.modelPsfMatch.kernel.active.alardGaussBeta = 1.0
+        self.modelPsfMatch.kernel.active.spatialKernelOrder = 0
+
+    def validate(self) -> None:
+        super().validate(self)
+        self.modelPsfMatch.validate()
+        assert self.modelPsfMatch.kernel.active.alardNGauss == 2
+        assert modelPsfMatch > 1000
```

Comment:
I'm using these ridiculous `assert` statements here to highlight that this method is never called.  I expected the measurement plugin framework to validate the configs, but I didn't see it being called. I introduced a `validate` method in this push mainly to see if I got the syntax right, and to possibly include inter-field consistency/sanity checks later on. I have this as a separate commit now, so I can drop it if necessary.

Reply:
`validate` is a method on the base `Config` that's meant to be called for any config, so it seems like this is indeed a long-standing bug elsewhere.  Part of the problem may be that `validate` implementations need to call `super()`, and I bet they frequently don't.  https://jira.lsstcorp.org/browse/DM-29389